Sources/PackageGraph/RepositoryPackageContainerProvider.swift

```diff
@@ -49,6 +47,19 @@ public class RepositoryPackageContainerProvider: PackageContainerProvider {
         // Create the container wrapper.
         return RepositoryPackageContainer(identifier: identifier, repository: repository, manifestLoader: manifestLoader)
     }
+
+    public func getContainer(for identifier: Container.Identifier, completion: @escaping (Result<Container, AnyError>) -> Void) {
+        repositoryManager.lookup(repository: identifier) { result in
+            do {
+                let handle = try result.dematerialize()
```

Comment:
Off topic, but I wonder if Result should have a monadic composition operation (like `map`) which either applies a block and returns a new result, or propagates the existing error.